### PR TITLE
WIP: Introduce a slim pvm backend using ssh

### PR DESCRIPTION
**Pull request by coolo**
This requires ssh access to a novalink installation, but currently does
little around it.

### CODE REVIEW

File: backend/pvm.pm [hunk 1]
```diff
@@ -24,6 +24,12 @@ use File::Basename;
 use Digest::MD5 'md5_hex';
 use osutils qw(dd_gen_params gen_params);
 
+# this backend relies on NovaLink tools being around on the worker
+# host. It supports HDD_1 and publishing assets
+
+# the spvm backend will only support the basics, but generally through
+# ssh to a novalink installation (so you only need ssh and terminal on worker)
+
```
Comment (okurz) [inline backend/pvm.pm]: POD?

File: backend/spvm.pm [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    my ($self) = @_;
+
+    # truncate the serial file
+    open(my $sf, '>', $self->{serialfile});
+    close($sf);
+
+    my $ssh = $testapi::distri->add_console('novalink-ssh', 'ssh-xterm', {
+        hostname => get_required_var('NOVALINK_HOSTNAME'),
+        password => get_required_var('NOVALINK_PASSWORD'),
+        username => get_var('NOVALINK_USERNAME', 'root')
```
Comment (ldevulder) [inline backend/spvm.pm]: You use `NOVALINK_USERNAME` but `NOVALINK_USER` in `run_cmd`.

**Comment by asdil12** [comment]
@coolo 
I rebased your patch and corrected the `NOVALINK_USER` to `NOVALINK_USERNAME`:
[909.patch](https://github.com/os-autoinst/os-autoinst/files/2122965/909.patch.txt)

Could you push this to your branch?

**Comment by coolo** [comment]
No, just pick my commits and make your own PR

**Comment by asdil12** [comment]
--> https://github.com/os-autoinst/os-autoinst/pull/979

**Pull request closed without merging**